File: ghc-4335c07/libraries/text/benchmarks/ruby/cut.rb
#!/usr/bin/env ruby

require './utils.rb'

def cut(filename, l, r)
  File.open(filename, 'r:utf-8') do |file|
    file.each_line do |line|
      puts line[l, r - l]
    end
  end
end

ARGV.each do |f|
  t = benchmark { cut(f, 20, 40) }
  STDERR.puts "#{f}: #{t}"
end

File: ghc-4335c07/libraries/text/benchmarks/python/cut.py
#!/usr/bin/env python

import utils, sys, codecs

def cut(filename, l, r):
    content = open(filename, encoding='utf-8')
    for line in content:
        print(line[l:r])

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: cut(f, 20, 40))
    sys.stderr.write('{0}: {1}\n'.format(f, t))
